### PR TITLE
Migrate the FSx networking validator

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
1. To access subnet_id in head_node and file_system_id in shared_storage, the validator has to be placed under their common parents. In this case, we put it in the cluster class.
2. For cross section validation, we need to only pass required fields instead of the whole sections to the validator. This is for the easiness of unit testing.
3. Migrating unit tests is not as easy as estimated. For example, there are some uncovered bug in our old unit tests. Probably the migration will speed up when I am more familiar to the code.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/tests/pcluster/config/test_validators.py [hunk 1]
```diff
@@ -1147,6 +1147,7 @@ def _head_bucket_stubber(mocker, boto3_stubber, bucket, num_calls):
             "only support using FSx file system that is in the same VPC as the stack",
         ),
         (  # not working case --> wrong ip permissions in security group
+            # FIXME this test is problematic. THe vpc is wrong, so the error should be relevant to vpc.
```
Comment (enrico-usai) [inline cli/tests/pcluster/config/test_validators.py:1150]: Let's talk about this.

File: cli/src/pcluster/models/cluster.py [hunk 2]
```diff
@@ -618,6 +619,14 @@ def __init__(
         self.iam = iam
         self.custom_actions = custom_actions
         self.cores = None
+        if self.shared_storage:
+            for storage in self.shared_storage:
+                if isinstance(storage, SharedFsx):
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/models/cluster.py:624]: The advantage of using `isinstance` over enum is that we have auto-complete function of the IDE for the variable `storage`

File: cli/src/pcluster/validators/cross_sections.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/validators/cross_sections.py]: Done

File: cli/tests/pcluster/validators/test_cross_section.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/tests/pcluster/validators/test_cross_section.py]: Done

File: cli/src/pcluster/validators/cluster.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (enrico-usai) [inline cli/src/pcluster/validators/cluster.py:1]: in a next patch I'd rename all the validators modules by adding `_validators` prefix.

validators/cluster.py --> validators/cluster_validators.py
etc

**Review by enrico-usai — APPROVED** [review]
LGTM great job!

**Pull request merged**